src/components/ContextMenu/components/CheckboxItem/CheckboxItem.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { forwardRef } from 'react';
+import { CheckboxItem as RadixDropdownMenuCheckboxItem } from '@radix-ui/react-dropdown-menu';
+import cx from 'classnames';
+
+import { useLevelContext } from '../../providers/LevelProvider';
+
+import { useContextMenuContext } from '../../ContextMenu.context';
+
+import type { CheckboxItemProps } from './CheckboxItem.props';
+
+import s from './CheckboxItem.module.css';
+
+const DISPLAY_NAME = 'ContextMenu.CheckboxItem';
+
+export const CheckboxItem = forwardRef<HTMLDivElement, CheckboxItemProps>(
+  (
+    {
+      className,
+      children,
+      icon,
```

Comment:
We need to be as close to the original implementation as possible to be more flexible.

<img width="1230" height="768" alt="image" src="https://github.com/user-attachments/assets/0f0aa531-15c9-4bb4-9d0d-d01ec11aef32" />